tsdb/engine/tsm1/compact.go

```diff
@@ -707,28 +695,18 @@ func (c *DefaultPlanner) findGenerations(skipInUse bool) tsmGenerations {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	last := c.lastFindGenerations
-	lastGen := c.lastGenerations
-
-	if !last.IsZero() && c.FileStore.LastModified().Equal(last) {
-		return lastGen
-	}
-
-	genTime := c.FileStore.LastModified()
 	tsmStats := c.FileStore.Stats()
 	generations := make(map[int]*tsmGeneration, len(tsmStats))
 	for _, f := range tsmStats {
-		gen, _, _ := c.ParseFileName(f.Path)
-
 		// Skip any files that are assigned to a current compaction plan
 		if _, ok := c.filesInUse[f.Path]; skipInUse && ok {
 			continue
 		}
 
-		group := generations[gen]
+		group := generations[f.Generation]
```

Comment:
Should we check that `f.Generation` and `f.Sequence` are not 0 before using them? This isn't an issue if we force supplying a parsing function or stop parameterizing it and always use the default.

Reply:
I believe we have to parameterize it to preserve tests which use unformatted TSM file names (from various mktemp sort of calls).